TugasAwesome/src/id/ac/itb/informatika/tugasawesome/model/GfPolynomial.java

```diff
@@ -56,15 +56,11 @@ private BigInteger randomCoeff(BigInteger prime) {
     /**
      * return value of {val} from polynomial {coeff}
      */
-    public BigInteger evaluatePolynomial(BigInteger val) {
-        BigInteger value = coeff.get(0);
-        
-        for (int i = 1; i<degree; i++) {
-            BigInteger curCoef = coeff.get(i);
-            BigInteger thisVal = curCoef.multiply(val.pow(i).mod(getPrime())).mod(getPrime());
```

Comment:
Dan di sini. Not exactly salah, tapi `val.pow(i)` itu bisa gede banget. Harusnya `val.modPow(i, getPrime())`